examples/zone-printer/README.md

```diff
@@ -0,0 +1,247 @@
+in-use and then extracting that context to its own file.
+
+```shell
+# First, do this for the us-east one.
+kubectl config use-context us-east
+kubectl config view  --minify --flatten > mciuseast
+
+# Second, do this for the eu-west one.
+kubectl config use-context eu-west
+kubectl config view --minify --flatten > mcieuwest
+
+# Finally, do this for the asia-east one.
+kubectl config use-context asia-east
+kubectl config view --minify --flatten > mciasiaeast
+```
+
+Combine (flatten) all of these extracted contexts to create a
+single kubeconfig file with just the contexts we care about
+
+```shell
+# Use KUBECONFIG to specify the context filenames we created in
+# the previous step
+KUBECONFIG=mciuseast:mcieuwest:mciasiaeast kubectl config view \
+  --flatten > mcikubeconfig
+```
+
+#### Creating new clusters
+
+You need at least two Kubernetes clusters in two different GCP zones
+to verify that `kubemci` works. Clusters must be v1.8.1 or newer.
+Let's create three GKE clusters and get their credentials for the
+purposes of this tutorial:
+
+```shell
+# Create a cluster in us-east and get its credentials
+gcloud container clusters create \
+    --cluster-version=1.8.1-gke.0 \
+    --zone=us-east4-a \
+    cluster-us-east
+KUBECONFIG=./mcikubeconfig gcloud container clusters get-credentials \
+    --zone=us-east4-a \
+    cluster-us-east
+
+# Create a cluster in eu-west and get its credentials
+gcloud container clusters create \
+    --cluster-version=1.8.1-gke.0 \
+    --zone=europe-west1-c \
+    cluster-eu-west
+KUBECONFIG=./mcikubeconfig gcloud container clusters get-credentials \
+    --zone=europe-west1-c \
+    cluster-eu-west
+
+# Create a cluster in asia-east and get its credentials
+gcloud container clusters create \
+    --cluster-version=1.8.1-gke.0 \
+    --zone=asia-east1-b \
+    cluster-asia-east
+KUBECONFIG=./mcikubeconfig gcloud container clusters get-credentials \
+    --zone=asia-east1-b \
+    cluster-asia-east
+```
+
+## 1. Deploy the application
+
+Deploy the application along with its NodePort service in each of the
+three clusters. You can get the cluster contexts from `kubectl config get-contexts` and iterate through all the clusters to deploy the
+application manifests. This could be accomplished by running the
+following loop:
+
+```shell
+for ctx in $(kubectl config get-contexts --kubeconfig=./mcikubeconfig -o name); do
+  kubectl --context="${ctx}" create -f app/
+done
+```
+
+## 2. Reserve a static IP on GCP.
+
+We need to reserve a static IP on GCP for kubemci. Run the following
+command to reserve the IP.
+
+```shell
+ZP_KUBEMCI_IP="zp-kubemci-ip" gcloud compute addresses create \
+    --global "${ZP_KUBEMCI_IP}"
+```
+
+## 3. Modify the Ingress manifest.
+
+Replace the value of `kubernetes.io/ingress.global-static-ip-name`
+annotation in `ingress/nginx.yaml` with the value of $ZP_KUBEMCI_IP.
+
+```shell
+sed -e "s/\$ZP_KUBEMCI_IP/${ZP_KUBEMCI_IP}/" ingress/nginx.yaml
```

Comment:
Why not just use "ip-for-kubemci" while reserving the IP. Why use zp-kubemci-ip first and then replace it?
Use zp-kubemci-ip or ip-for-kubemci in the yaml in the first place?

Reply:
You could. The idea is to teach people what's going on by making them do it.